app/scripts/components/exploration/timeline-head.tsx

```diff
@@ -0,0 +1,200 @@
+        document.body.style.cursor = 'grabbing';
+        select(this).attr('cursor', 'grabbing');
+      })
+      .on('drag', function dragged(event) {
+        if (event.x < 0 || event.x > width) {
+          return;
+        }
+
+        const dx = event.x - event.subject.x;
+        const currPos = xScaled(selectedDay);
+        const newPos = currPos + dx;
+
+        const dateFromPos = startOfDay(xScaled.invert(newPos));
+
+        const [start, end] = domain;
+        const interval = { start, end };
+
+        const newDate = clamp(dateFromPos, interval);
+
+        if (selectedDay.getTime() !== newDate.getTime()) {
+          onDayChange(newDate);
+        }
+      })
+      .on('end', function dragended() {
+        document.body.style.cursor = '';
+        select(this).attr('cursor', 'grab');
+      });
+
+    select(rectRef.current).call(dragger);
+  }, [width, domain, selectedDay, onDayChange, xScaled]);
+
+  return (
+    <TimelineHeadSVG width={width}>
+      <line
+        x1={xScaled(selectedDay)}
+        x2={xScaled(selectedDay)}
+        y1={0}
+        y2='100%'
+        stroke={theme.color?.base}
+      />
+      <g transform={`translate(${xScaled(selectedDay)}, 0)`} ref={rectRef}>
+        {children}
+      </g>
+    </TimelineHeadSVG>
+  );
+}
+
+export function TimelineHeadP(props: Omit<TimelineHeadProps, 'children'>) {
+  const theme = useTheme();
+
+  return (
+    <TimelineHead {...props}>
+      <path
+        transform='translate(-14, -4)'
+        d='M4 14.6459C4 15.4637 4.4979 16.1992 5.25722 16.5029L13.2572 19.7029C13.734 19.8936 14.266 19.8936 14.7428 19.7029L22.7428 16.5029C23.5021 16.1992 24 15.4637 24 14.6459L24 6C24 4.89543 23.1046 4 22 4L6 4C4.89543 4 4 4.89543 4 6L4 14.6459Z'
+        fill={theme.color?.surface}
+        stroke={theme.color?.['base-200']}
+        style={{
+          filter: dropShadowFilter,
+          pointerEvents: 'all',
+          cursor: 'grab'
+        }}
+      />
+      <text fill={theme.color?.base} fontSize='0.75rem' y='0' x='-4' dy='1em'>
+        P
+      </text>
+    </TimelineHead>
+  );
+}
+
+export function TimelineHeadL(props: Omit<TimelineHeadProps, 'children'>) {
+  const theme = useTheme();
+
+  return (
+    <TimelineHead {...props}>
+      <path
+        transform='translate(-6, -4)'
+        d='M4 6C4 4.89543 4.89543 4 6 4H15.1716C15.702 4 16.2107 4.21071 16.5858 4.58579L22.5858 10.5858C23.3668 11.3668 23.3668 12.6332 22.5858 13.4142L16.5858 19.4142C16.2107 19.7893 15.702 20 15.1716 20H6C4.89543 20 4 19.1046 4 18V6Z'
+        fill={theme.color?.surface}
+        stroke={theme.color?.['base-200']}
+        style={{
+          filter: dropShadowFilter,
+          pointerEvents: 'all',
+          cursor: 'grab'
+        }}
+      />
+      <text fill={theme.color?.base} fontSize='0.75rem' y='0' x='2' dy='1em'>
+        L
+      </text>
+    </TimelineHead>
+  );
+}
+
+export function TimelineHeadR(props: Omit<TimelineHeadProps, 'children'>) {
```

Comment:
You could wrap TimelineHeadP/L/R into a single component passing down the letter and the path object.

Reply:
There are several properties that are changing, not only the path (transform, text position, etc)